File: 백준/Gold/1931. 회의실 배정/회의실 배정.py
import sys
input = sys.stdin.readline

n = int(input())
meetings = []

for _ in range(n):
    s, e = map(int, input().split())
    meetings.append((s, e))

meetings.sort(key=lambda x: (x[1], x[0]))

count = 0
last_end = 0

for start, end in meetings:
    if start >= last_end:
        count += 1
        last_end = end

print(count)

File: 백준/Gold/1931. 회의실 배정/회의실 배정.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');
const n = Number(input[0]);
const meetings = [];

for (let i = 1; i <= n; i++) {
    const [s, e] = input[i].split(' ').map(Number);
    meetings.push([s, e]);
}

meetings.sort((a, b) => {
    if (a[1] !== b[1]) {
        return a[1] - b[1];
    }
    return a[0] - b[0];
});

let count = 0;
let lastEnd = 0;

for (const [start, end] of meetings) {
    if (start >= lastEnd) {
        count++;
        lastEnd = end;
    }
}

console.log(count);